src/components/Stepper/index.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from "react";
+import Button from "../Button";
+import Icon from "../Icon";
+
+const Stepper = ({
+  steps,
+  isDisableFinish = false,
+  onFinish = () => {},
+  onStepChange = () => {},
+  finishBtnText = "Finish",
+}) => {
+  const [currentStep, setCurrentStep] = useState(0);
+
+  useEffect(() => {
+    const initialStep = steps.findIndex((step) => step.isInitial);
+    setCurrentStep(initialStep !== -1 ? initialStep : 0);
+  }, [steps]);
+
+  const handleNext = () => {
+    if (currentStep < steps.length - 1) {
+      const nextStep = currentStep + 1;
+      setCurrentStep(nextStep);
+      onStepChange(steps[nextStep]);
```

Comment:
I think this onStepChange, should accept a parameter  a function (or a function should be one of the optiosn). If this function evaluates to true, then the step is changed. This way it will allow me in the code to write some validation, and then the step to change. @ifrim wdyt?

Reply:
Yeah, I would `await` on the response of the `onStepChange` call. This way you can make sync or async validations in the function. And then go ahead and increment the step only if the awaited value is truthy.

Reply:
I see, what do you think about doing it based on the `props` 

example
```js

<Stepper 
allowNextStep={false}
activeStepClass="step-error`

/>

```

@ifrim @lghiur

Reply:
The way I see it is that the problem that needs to be solved is that the user interacts with the component and then a decision needs to be made if the step is changed or not (in which case an error may or may not be displayed).
The simplest API I can think of for this is using a callback called at the moment of the user interaction and doing stuff based on the response.

The validations need to happen at the moment the user requested the step change. In order to do this with the props based approach then the parent component will need to know the next step and the moment the user interacted with the component requesting the step change. It will also need to keep track of the current step and transition the value of `allowNextStep` when the step changes and while the validations are running. This is boilerplate code that will need to be replicated every time you would use the component. I would rather have this encapsulated inside the component. It will also be easier to do future changes if everything that is not specific to a use case is inside the component instead of the outside.

Having said that not sure if I completely understand the use cases for this component so whatever you think is best.